### PR TITLE
WIP: Add supported platforms handling

### PR DESCRIPTION
**Pull request by RamblingCookieMonster**
Starting to work on xplat support (#11).  One thing we'll need is a way to narrow down which dependency scripts support which platforms.  This is an initial, minimal-thought take on adding this to the PSDependMap schema and using that data in a few spots...

* Add supports <array of platforms> to PSDependMap
* Add supports, and supported properties to Get-PSDependType output
* Skip invocation of dependency scripts that do not support the current platform

Currently we do not fail all dependencies, we simply skip unsupported dependencies - open to failing all the things instead

Hit me up if you have any ideas or concerns!  Will leave this open a few days.

### CODE REVIEW

**Comment by Dargmuesli** [comment]
What about a warning for unsupported dependency types? But failing makes more sense, I think. What is the purpose of dependencies when they cannot be downloaded...

**Comment by RamblingCookieMonster** [comment]
Down the line, will need to add support for platforms to requirement/depend files, similar to [bundler/gemfiles](http://bundler.io/man/gemfile.5.html#PLATFORMS)

**Comment by RamblingCookieMonster** [comment]
Hiyo @michaeltlombardi!

Looking through [npm.ps1](https://github.com/RamblingCookieMonster/PSDepend/blob/master/PSDepend/PSDependScripts/Npm.ps1), looks like it might already support xplat, barring a switch from mkdir to new-item?  I'm assuming [these params](https://github.com/RamblingCookieMonster/PSDepend/blob/master/PSDepend/Private/Get-NodeModule.ps1) aren't platform specific to npm?

Cheers!

**Comment by Dargmuesli** [comment]
No, sorry. I don't have access to any macOS devices.

**Comment by Dargmuesli** [comment]
But looking over it again: it **should** be compatible with mac. There is room for improvement though, as e.g. privilege detection cannot be realized on non-Windows OSs right now (afaik) and so on.

**Comment by RamblingCookieMonster** [comment]
Cool!  will test it out, but yeah, don't see anything at a glance - thanks!

**Comment by RamblingCookieMonster** [comment]
Alrighty... so this is in a good-enough-but-probably-has-a-bunch-of-bugs stage, in terms of cross platform support.  Going to merge it in so that I can start using it for a few things, but this really needs more work before tagging it or officially saying it has cross platform support

I'll likely get antsy and merge this Sunday night or Monday AM, barring any major issues

cc folks who have mentioned xplat - @gaelcolas @Dargmuesli @gjonespf

File: PSDepend/PSDependMap.psd1 [hunk 1]
```diff
@@ -2,62 +2,73 @@
...
 
     Git = @{
         Script = 'Git.ps1'
         Description = 'Clone a git repository'
+        Supports = 'windows'
     }
 
     GitHub = @{
         Script = 'GitHub.ps1'
         Description = 'EXPERIMENTAL: Download and extract a GitHub repo'
```
[the remark this answers is not in the document]
Comment (RamblingCookieMonster) [reply PSDepend/PSDependMap.psd1]: Oh!  Thanks for the reminder, will do

**Comment by gjonespf** [comment]
TY for the heads up.  Will definitely look at making use of this, have to experiment once merged.  My primary use previously was for xplat build, but have shifted to using Cake.  This will still be super useful for xplat deployment scripts I think.  I should whack this in a Docker container for testing.

File: PSDepend/PSDepend.psd1 [hunk 2]
```diff
@@ -4,7 +4,7 @@
 RootModule = 'PSDepend.psm1'
 
 # Version number of this module.
-ModuleVersion = '0.1.63'
+ModuleVersion = '0.2.0'
```
Comment (Dargmuesli) [inline PSDepend/PSDepend.psd1]: Are you going to release a release candidate or beta?

Comment (RamblingCookieMonster) [reply PSDepend/PSDepend.psd1]: Hiyo!

Probably not - thought process:

* This just adds a bit to validate platform support of independent scripts
* We're not saying 'xplat is a thing!' - will just mention that it works in some very limited, specific environments and that we'll update and ping #11 as we work on xplat support and functionality
* I don't have time : )  Makes sense for a large project like Pester, but I barely have time to give the okay for other folks' code

It's definitely a cool idea, but for this particular PR I'm happy enough to just bump the version

Cheers!

Comment (Dargmuesli) [reply PSDepend/PSDepend.psd1]: The just merged changes are now available via PSGallery as version 0.1.66 and not 0.2.0. Is that correct?

Comment (RamblingCookieMonster) [reply PSDepend/PSDepend.psd1]: Correct but unintended : )  Using different build setups per module sometimes bites me, assumed I had the psd1-can-override-version-bump login in place... going to add some release notes and bump for a release now

Comment (RamblingCookieMonster) [reply PSDepend/PSDepend.psd1]: Whew!  worked - https://www.powershellgallery.com/packages/PSDepend/0.2.0

Comment (RamblingCookieMonster) [reply PSDepend/PSDepend.psd1]: Probably completely unnecessary, but unlisted 0.1.66, 0.1.67 that had these changes without the version bump, things should be in order now.

Cheers!

Comment (Dargmuesli) [reply PSDepend/PSDepend.psd1]: Cool!

**Pull request merged**